### PR TITLE
Add API Resource based scope metadata service

### PR DESCRIPTION
**Pull request by SujanSanjula96**
### Proposed changes in this pull request

Add DefaultScopeMetadata service for Role V2 implementation with new API Resource implementation to get scope metadata like display name and description grouped by API resources. This is needed to display user friendly names for scopes in consent page.

### When should this PR be merged
Should merge https://github.com/wso2/carbon-identity-framework/pull/5003 and bump the framework version.

Related Issue - https://github.com/wso2/product-is/issues/16973

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java [hunk 1]
```diff
@@ -1188,4 +1192,34 @@ protected void unsetRealmService(RealmService realmService) {
...
+            name = "api.resource.mgt.service.component",
+            service = APIResourceManager.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetAPIResourceManager")
+    protected void setAPIResourceManager(APIResourceManager apiResourceManager) {
+
+        log.debug("Setting the API Resource Manager.");
+        OAuth2ServiceComponentHolder.getInstance()
+                .setAPIResourceManager(apiResourceManager);
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java:1211]: can't these two lines get merged into one line?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java [hunk 2]
```diff
@@ -1188,4 +1192,34 @@ protected void unsetRealmService(RealmService realmService) {
...
+    /**
+     * Unset API Resource manager service.
+     *
+     * @param apiResourceManager API Resource Manager.
+     */
+    protected void unsetAPIResourceManager(APIResourceManager apiResourceManager) {
+
+        log.debug("Unsetting the API Resource Manager.");
+        OAuth2ServiceComponentHolder.getInstance()
+                .setAPIResourceManager(null);
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java:1223]: can't these two lines get merged into one line

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java [hunk 3]
```diff
@@ -1188,4 +1192,34 @@ protected void unsetRealmService(RealmService realmService) {
...
+     */
+    @Reference(
+            name = "api.resource.mgt.service.component",
+            service = APIResourceManager.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetAPIResourceManager")
+    protected void setAPIResourceManager(APIResourceManager apiResourceManager) {
+
+        log.debug("Setting the API Resource Manager.");
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java:1209]: move the debug log after actual setter method, change the log to indicate "Set the API resource manager in to `<xyz>` bundle"

Do the same in unset method

File: components/org.wso2.carbon.identity.oauth/pom.xml [hunk 4]
```diff
@@ -324,6 +324,10 @@
             <groupId>org.wso2.carbon.identity.framework</groupId>
             <artifactId>org.wso2.carbon.identity.central.log.mgt</artifactId>
         </dependency>
+        <dependency>
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.oauth/pom.xml:327]: change the year range in license

File: pom.xml [hunk 5]
```diff
@@ -678,6 +678,11 @@
                 <artifactId>org.wso2.carbon.identity.central.log.mgt</artifactId>
                 <version>${carbon.identity.framework.version}</version>
             </dependency>
+            <dependency>
```
Comment (AnuradhaSK) [inline pom.xml:681]: change the year range in license

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java (hunk 1 above)
Comment (SujanSanjula96) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java]: Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2196/commits/1f6b9ab57f76c33caa73487e50b8cab7a2c3a317

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java (hunk 2 above)
Comment (SujanSanjula96) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java]: Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2196/commits/1f6b9ab57f76c33caa73487e50b8cab7a2c3a317

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java (hunk 3 above)
Comment (SujanSanjula96) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java]: Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2196/commits/1f6b9ab57f76c33caa73487e50b8cab7a2c3a317

File: components/org.wso2.carbon.identity.oauth/pom.xml (hunk 4 above)
Comment (SujanSanjula96) [reply components/org.wso2.carbon.identity.oauth/pom.xml:327]: Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2196/commits/1f6b9ab57f76c33caa73487e50b8cab7a2c3a317

File: pom.xml (hunk 5 above)
Comment (SujanSanjula96) [reply pom.xml:682]: Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2196/commits/1f6b9ab57f76c33caa73487e50b8cab7a2c3a317

**Review by thanujalk — APPROVED** [review]

**Comment by SujanSanjula96** [comment]
Locally ran the failed integration test and verified.

Screenshot 2023-10-27 at 09 03 07

**Pull request merged**